pkg/catalog/manager/traverse.go

```diff
@@ -129,7 +147,28 @@ func traverseFiles(repoPath string, catalog *v3.Catalog) ([]v3.Template, []strin
 		template.Spec.CatalogID = catalog.Name
 		template.Name = fmt.Sprintf("%s-%s", catalog.Name, template.Spec.FolderName)
 
-		templates = append(templates, template)
+		v3.CatalogConditionRefreshed.Unknown(catalog)
+		v3.CatalogConditionRefreshed.Message(catalog, fmt.Sprintf("syncing template %v", template.Name))
+		if newCatalog, err := m.catalogClient.Update(catalog); err != nil {
```

Comment:
is the error handling correct here? seems like you change the value of catalog but otherwise do not handle the error

Reply:
i ll change it to:
```
if newCatalog, err := m.catalogClient.Update(catalog); err == nil {
			catalog = newCatalog
		} else {
			catalog, _ = m.catalogClient.Get(catalog.Name, metav1.GetOptions{})
		}
```
this is to mimic the logic in here https://github.com/rancher/rancher/blob/master/pkg/controllers/management/node/utils.go#L173-L178 darren wrote this for updating transitioning msg. The idea is to ignore the update error and always refresh the object.